internal/webserver/monitoring.go

```diff
@@ -0,0 +1,41 @@
+package webserver
+
+import (
+	"github.com/neutron-org/neutron-query-relayer/internal/metrics"
+	"net/http"
+
+	nlogger "github.com/neutron-org/neutron-logger"
+	"github.com/neutron-org/neutron-query-relayer/internal/relay"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+
+	"go.uber.org/zap"
+)
+
+const MonitoringLoggerContext = "monitoring"
+
+type PromWrapper struct {
+	promHandler http.Handler
+	storage     relay.Storage
+	logger      *zap.Logger
+}
+
+func NewPromWrapper(logRegistry *nlogger.Registry, storage relay.Storage) PromWrapper {
+	return PromWrapper{
+		promHandler: promhttp.Handler(),
+		storage:     storage,
+		logger:      logRegistry.Get(MonitoringLoggerContext),
+	}
+}
+
+func (p PromWrapper) FillUnsuccessfulTxsMetric() {
+	txs, err := p.storage.GetAllUnsuccessfulTxs()
+	if err != nil {
+		p.logger.Error("failed to get unsuccessful txs from storage", zap.Error(err))
+	}
+	metrics.SetUnsuccessfulTxsSizeQueue(len(txs))
+}
+
+func (p PromWrapper) ServeHTTP(res http.ResponseWriter, req *http.Request) {
+	p.FillUnsuccessfulTxsMetric()
```

Comment:
what's the meaning of this call in this method?

Reply:
I can see two ways to achieve the goal to get amount of unsuccessful txs:
1) eval ` SetUnsuccessfulTxsSizeQueue` on every `saveIntoErrorQueue` and `removeFromErrorQueue`, but i do not feel it right to inject the global metrics into the storage, may be i'm wrong, we already have the metrics spread throughout the code
2) measure the amount of txs during the `/metrics` request

Reply:
re 2:
you now eval `SetUnsuccessfulTxsSizeQueue` twice on each request: first in `ServeHTTP` since it's called on each incoming request, and second in the `UnsuccessfulTxs` since it's the `/metrics` endpoint handler. so, at least you can delete the `FillUnsuccessfulTxsMetric` call and the func itself and keep calculating only in the `metrics` handler.

re 1:
I find this solution as the best option here. I'm totally fine with importing metric setters across the app, I believe it's the way they are supposed to be used

Reply:
i've changed a mind about first approach.
you need an extra storage-metric specific code to determine is it a new key-value or a modification existing one https://github.com/neutron-org/neutron-query-relayer/blob/feat/query-failed-txs/internal/storage/leveldb.go#L231. I'm against extra code in the storage only for metrics (`Has` method)

Reply:
re 2: sorry, I'm wrong, ignore
re 1: okay, I see, it's not that easy indeed

I'm fine with the current version, don't think it's gonna hurt or slow the app down in any way